### PR TITLE
New endpoints for annotation results

### LINKED ISSUE #88: Endpoint for annotation results
for review page we need 2 more endpoints, something like:

`GET /exp/{id}/file-pairs` list of file pairs:
```
[
{id: 1, path: ... },
{id: 1, path: ... }
]
```

`GET /exp/{id}/file-pairs/{id}/annotations`

it should return json with user annotations, something like:
```
data: {
  yes: 1,
  no: 4,
  maybe: 2,
  skip: 0
}
```

### PR DESCRIPTION
**Pull request by carlosms**
Fixes #88.

There are two new routes:
- `experiments/{id}/file-pairs`
- `experiments/{id}/file-pairs/{id}/annotations`

Comments:
- Authorization (restrict to Requesters only) is left out of the scope of this PR, it will be addressed later.
- `serializer.simplifiedFilePairResponse` uses `leftPath` & `rightPath` to be consistent with `filePairResponse.left/rightBlobId`. But we may want to change it to a json with `left.Path` and `left.BlobId`

### CODE REVIEW

File: server/serializer/serializers.go [hunk 1]
```diff
@@ -122,6 +138,22 @@ func NewFilePairResponse(fp *model.FilePair) *Response {
 		fp.ID, fp.Diff, fp.Score, fp.Left.BlobID, fp.Right.BlobID})
 }
 
+type simplifiedFilePairResponse struct {
```
Comment (smacker) [inline server/serializer/serializers.go]: do you think `listFilePairResponse` would be a better name?

File: server/repository/common.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package repository
+
+// scannable is used to call .Scan for both sql.Row and sql.Rows
+type scannable interface {
```
Comment (smacker) [inline server/repository/common.go:4]: https://golang.org/pkg/database/sql/#Scanner ?

File: server/handler/assignments.go [hunk 3]
```diff
@@ -96,3 +96,45 @@ func SaveAssignment(repo *repository.Assignments) RequestProcessFunc {
...
+		if err != nil {
+			return nil, err
+		}
+
+		pairID, err := urlParamInt(r, "pairId")
+		if err != nil {
+			return nil, err
+		}
+
+		assignments, err := repo.GetByExperimentPair(experimentID, pairID)
```
Comment (smacker) [inline server/handler/assignments.go:114]: hm... it's not `assignments`, is it?

**Review by smacker — CHANGES_REQUESTED** [review]
great work! just few small comments

File: server/repository/common.go (hunk 2 above)
Comment (carlosms) [reply server/repository/common.go:4]: I believe that is the interface used by Rows.Scan, to insert individual values into each argument.

File: server/handler/assignments.go (hunk 3 above)
Comment (carlosms) [reply server/handler/assignments.go:114]: Yes, those come from the assignments table

File: server/repository/common.go (hunk 2 above)
Comment (smacker) [reply server/repository/common.go:4]: Sorry, don't understand you. https://golang.org/pkg/database/sql/#Scanner is absolutely the same as this one.

File: server/handler/assignments.go (hunk 3 above)
Comment (smacker) [reply server/handler/assignments.go:114]: ah. got you. thanks!

File: server/repository/common.go (hunk 2 above)
Comment (smacker) [reply server/repository/common.go:4]: ah. it's actually different. 👍

**Review by smacker — APPROVED** [review]

**Review by bzz — APPROVED** [review]
LGTM

**Pull request merged**